src/components/form/Input.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { InputHTMLAttributes } from 'react'
+
+type InputProps = {
+  label?: string
+  isInvalid?: boolean
+  isInvalidBlur?: boolean
+  trailingText?: string
+  leadingText?: string
+  errorMessage?: string
+  tooltip?: string
+}
+
+const Input: React.FC<InputHTMLAttributes<HTMLInputElement> & InputProps> = ({
+  label,
+  isInvalid,
+  isInvalidBlur,
+  trailingText,
+  leadingText,
+  errorMessage,
+  tooltip,
+  ...props
+}) => {
+  const [focus, setFocus] = React.useState(false)
+  const [showError, setShowError] = React.useState(false)
+  const baseStyles = 'px-3 py-2 outline-none w-full lg:text-base'
+  const baseContainerStyles = 'flex group rounded-lg overflow-hidden'
+  const shadowStyles =
+    isInvalid || (isInvalidBlur && showError)
+      ? 'shadow-red-outline'
+      : 'shadow-dropdown focus-within:shadow-blue-outline dark:shadow-dark-dropdown dark:focus-within:shadow-dark-blue-outline'
+  const borderStyles =
+    isInvalid || (isInvalidBlur && showError)
+      ? 'border border-red-500'
+      : 'border border-gray-300 dark:border-gray-500 focus-within:border-blue-500 dark:focus-within:border-blue-500'
+  const backgroundStyles = 'bg-white dark:bg-gray-medium'
+  const textStyles = 'dark:text-white'
+
+  const inputElement = (
+    <input
+      className={`${baseStyles} ${backgroundStyles} ${textStyles}`}
+      {...props}
+      onFocus={() => {
```

Comment:
return value typing ?

Reply:
The return value is inferred by Typescript here! Besides, I don't think it would be meaningful to explicit because that function isn't going to be reused anywhere. WDYT?